### ISSUE TITLE
Subdivisions and their relation to BatchNorm

### ISSUE THREAD

**Issue by vishnubanna** [pjreddie/darknet#2393]
quick question, how is the gradients aggregated across subdivisions, are the gradients for each batch saved and then averaged, summed together or some other operation. At 64 batch size with 8 sub divisions, means 8 independent gradients. along the same lines, are the batch norm statistics updated every 8 subdivisions as well in order to ensure that both the weight and mean updates happen at the same cycle? and finally for the batch norm given that you are using 8 subdivisions, then you cannot guarantee the net gradient is the same as if you were to take only 1 subdivision as each batch is forward propagated against its own mean and variance, so the 64/1 batch must be different in over all characteristic than the 64/8 correct? 

I am sure most of this is obvious, i just need to make sure i am not going insane and making some error in understanding.
